builds/jenkins-build-pack-api/Jenkinsfile

```diff
@@ -41,430 +42,460 @@ import groovy.transform.Field
-      }
-      events.sendCompletedEvent('UPDATE_ENVIRONMENT', 'Environment update event for deployment completion', environmentDeploymentMetadata.generateEnvironmentMap("deployment_completed", environment_logical_id, env_metadata), environment_logical_id)
-      events.sendCompletedEvent('UPDATE_DEPLOYMENT', "Deployment completion event for ${env_key} deployment", environmentDeploymentMetadata.generateDeploymentMap("successful", environment_logical_id, gitCommitHash), environment_logical_id)
-      events.sendCompletedEvent('DEPLOY_TO_AWS', 'Successfully deployed services to AWS', serviceContext, environment_logical_id)
-    }//stage of deployment to an environment ends here
-
-  }//dir ends here
+        events.sendCompletedEvent('UPDATE_ENVIRONMENT', 'Environment update event for deployment completion', environmentDeploymentMetadata.generateEnvironmentMap("deployment_completed", environment_logical_id, env_metadata), environment_logical_id)
+        events.sendCompletedEvent('UPDATE_DEPLOYMENT', "Deployment completion event for ${env_key} deployment", environmentDeploymentMetadata.generateDeploymentMap("successful", environment_logical_id, gitCommitHash), environment_logical_id)
+        events.sendCompletedEvent('DEPLOY_TO_AWS', 'Successfully deployed services to AWS', serviceContext, environment_logical_id)
+      }//stage of deployment to an environment ends here
+
+      //Udating safe details
+        if (config['domain'] != 'jazz' && configLoader.TVAULT && configLoader.TVAULT.IS_ENABLED) {
+          stage('Update Safe') {
+            events.sendStartedEvent('UPDATE_SAFE', 'safe updation started', context_map, environment_logical_id)
+            withCredentials([
+            [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: accountDetails.CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+              try {
+                // initialize aws credentials
+                def randomString = utilModule.generateRequestId();
+                credsId = "jazz-${randomString}";
+                sh "aws configure set profile.${credsId}.region ${service_config.region}"
+                sh "aws configure set profile.${credsId}.aws_access_key_id $AWS_ACCESS_KEY_ID"
+                sh "aws configure set profile.${credsId}.aws_secret_access_key $AWS_SECRET_ACCESS_KEY"
+            
+                def environment = environmentDeploymentMetadata.getEnvironmentInfo(environment_logical_id)
+                if (environment && environment.metadata && environment.metadata.safe_details) {
+                  def safeName = environment.metadata.safe_details.name
+                  def updatedIamRoleARN = vaultUtilityModule.updateSafeDetails(safeName, lambdaARN, credsId)
+                  events.sendCompletedEvent('CREATE_ASSET', null, utilModule.generateAssetMap(config['provider'], updatedIamRoleARN, "iam_role", config), environment_logical_id);
```

Comment:
We will need to send CREATE_ASSET only when the platform creates the role. If user supplied custom role, we don't track it as an asset that the platform creates.